Base/src/IPushQueueStorage.cs

```diff
@@ -36,8 +38,8 @@ public interface IPushQueueStorage : IQueueStorage
   /// <returns>
   ///   Task representing the asynchronous execution of the method
   /// </returns>
-  public Task PushMessagesAsync(IEnumerable<string> messages,
-                                string              partitionId,
-                                int                 priority          = 1,
-                                CancellationToken   cancellationToken = default);
+  public Task PushMessagesAsync(IEnumerable<MessageData> messages,
+                                string                   partitionId,
+                                int                      priority          = 1,
+                                CancellationToken        cancellationToken = default);
```

Comment:
I wonder. Should the `priority` and `partitionId` be kept as their own parameters? or integrated directly into the `MessageData` structure? I think I would prefer the latter.

@aneojgurhem What do you think?

Reply:
It will change the meaning of the interface and delegate the management of the priority/partition to the implementation. BTW, these informations are already present in MessageData.TaskOptions